pkg/controller/statefulset/stateful_set_control.go

```diff
@@ -132,6 +132,13 @@ func (ssc *defaultStatefulSetControl) UpdateStatefulSet(set *apps.StatefulSet, p
 		if !isCreated(replicas[i]) {
 			return ssc.podControl.CreateStatefulPod(set, replicas[i])
 		}
+		// If we find a Pod that is currently terminating, we must wait until graceful deletion
+		// completes before we continue to make  progress.
+		if isTerminating(replicas[i]) {
+			glog.V(2).Infof("StatefulSet %s is waiting for Pod %s to Terminate",
```

Comment:
This is debugging info - our prod installations run with v(2) and it's unlike that admins will care about these logs. Anyway, since v(2) is used across the controller, we can discuss in a follow-up issue.